src/checkers/anityachecker.py

```diff
@@ -55,7 +56,10 @@ async def check(self, external_data: ExternalBase):
                 versions = result["versions"]
             if constraints:
                 versions = filter_versions(versions, constraints, sort=False)
-            latest_version = versions[0]
+            try:
+                latest_version = versions[0]
+            except IndexError as err:
+                raise CheckerQueryError("Can't find matching version") from err
```

Comment:
Might be nice to give more details about why not:

```suggestion
                raise CheckerQueryError(f"Can't find matching {'stable ' if stable_only else ''}version{ f' satisfying {constraints}' if constraints else ''}") from err
```

but having written that, it's really hard to read, and I guess it will be obvious by looking at the manifest.

Reply:
>I guess it will be obvious by looking at the manifest.

Yeah. Also `constraints` will look messy when printed, since it's not just a `dict` but either `OrderedDict` or `CommentedMap`.